test/test_nn.py

```diff
@@ -7743,6 +7743,47 @@ def test_smoothl1loss_negative_beta_not_supported(self):
         with self.assertRaises(RuntimeError):
             F.smooth_l1_loss(torch.randn(2, 2), torch.randn(2, 2), beta=-1.0)
 
+    def test_smooth_l1_loss_vs_huber_loss(self):
```

Comment:
Make this a device-generic test by putting it in the device generic test class and accepting the device argument.